File: ExcelDivider/src/main/java/ru/tsystems/divider/service/api/FeatureService.java
package ru.tsystems.divider.service.api;

import ru.tsystems.divider.entity.Feature;

public interface FeatureService {
    /**
     * Find One param entity by this param.
     *
     * @param param
     *            param for searching.
     * @param nature
     * @return Entity if it exist, either null.
     */
    public Feature findByParam(String param, String nature);

    /**
     * Persist into db OneParamEntity.
     *
     * @param entity
     *            One param Entity.
     */
    public void persist(Feature entity);

    public Feature createFeatur(String title, String nature);
}

File: ExcelDivider/src/main/kt/ru/tsystems/divider/service/api/FeatureService.kt
package ru.tsystems.divider.service.api

import ru.tsystems.divider.entity.Feature

interface FeatureService {
    /**
     * Find One param entity by this param.
     *
     * @param param
     * param for searching.
     * @param nature
     * @return Entity if it exist, either null.
     */
    fun findByParam(param: String, nature: String): Feature?

    /**
     * Persist into db OneParamEntity.
     *
     * @param entity
     * One param Entity.
     */
    fun persist(entity: Feature)

    fun createFeatur(title: String, nature: String): Feature
}